drivers/FlashIAP.h

```diff
@@ -49,17 +49,29 @@ extern uint32_t Load$$LR$$LR_IROM1$$Limit[];
 
 namespace mbed {
 
-/** \addtogroup drivers */
+/** \ingroup drivers */
+/** \addtogroup drivers-public-api */
+/** @{*/
+/**
+ * \defgroup drivers_FlashIAP FlashIAP class
+ * @{
+ */
 
 /** Flash IAP driver. It invokes flash HAL functions.
  *
  * @note Synchronization level: Thread safe
- * @ingroup drivers
  */
 class FlashIAP : private NonCopyable<FlashIAP> {
 public:
-    FlashIAP();
-    ~FlashIAP();
+    FlashIAP()
```

Comment:
The implementation for the memeber functions in this file should be left in the .cpp

Reply:
See my overall comment here: https://github.com/ARMmbed/mbed-os/pull/10955#issuecomment-509937556

Reply:
In addition, in this particular case the method (constructor) is empty thus resulting in a pointless call with the overhead of calling the method. Making them inline removes that unnecessary overhead.